File: java/src/main/java/gov/doc/ntia/sigmf/util/AnnotationComparator.java
package gov.doc.ntia.sigmf.util;



import gov.doc.ntia.sigmf.Annotation;

import java.util.Comparator;

public class AnnotationComparator<T extends Annotation> implements Comparator<Annotation> {
    @Override
    public int compare(Annotation o1, Annotation o2) {
        return o1.getSampleStart().compareTo(o2.getSampleStart());
    }
}

File: Csharp/sigmf/sigmf/util/AnnotationComparator.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Text;

namespace sigmf
{
    public class AnnotationComparator<T> : IComparer<Annotation>
    {
        public int Compare(Annotation a1, Annotation a2) {
            return a1.SampleStart.CompareTo(a2.SampleStart);
        }

    }
}
